app/services/badge_granter.rb

```diff
@@ -104,6 +104,27 @@ def self.revoke(user_badge, options = {})
     end
   end
 
+  def self.revoke_all(badge)
+    custom_badge_name = TranslationOverride.where(translation_key: badge.translation_key).pluck(:value)
+    custom_badge_name = custom_badge_name.reduce("") { |m, v| m += "'#{v}'," }.delete_suffix(',')
```

Comment:
Won't this fail if there's a single quote in the translation? We should probably re-write this query to use active record and sanitization.